articles/frontend-is-not-your-enemy.md

```diff
@@ -0,0 +1,157 @@
+
+Nonetheless, many good things have come out of this wormhole...
+
+## ECMAScript and OOP
+
+ECMAScript is a standard which drives all new native JavaScript features. ES6 specifically, brought many improvements which developers craved for a long time.
+
+Classes. There have been several patterns used to simulate behavior of classes we are so fond of in OOP languages such as Java or C#. There is no longer need for that cold sweat before we use Prototype or function constructors, hoping chaos won't unfold in process.
+
+```js
+class MyObject {
+
+    constructor(number) {
+        this.field = number;
+    }
+
+    // property
+    get square() {
+        return this.field ** 2;
+    }
+
+    // method
+    inc() {
+        this.field++;
+    }
+
+}
+
+let foo = new MyObject(42);
+
+foo.inc();
+
+console.log(foo.square); // 1849
+```
+
+There are many new features which you will fall in love with. Like *let, const, lambdas, destructuring, async/await, default params...*
+
+It is highly advised to look into ES6 documentation and get familiar with the features. Visit <http://es6-features.org>
+
+## Compile Time Checking
+
+Ok, I know who you are... You are one of those people who call themselves *back-end* developer. And you probably worked in strongly typed languages, and you are used to having your IDE insult you with ***Cannot assign boolean to String!***  So you find frontend rather disgusting.
+
+Well guess what, you have it all now! You have it with TypeScript (or Flow, KotlinJS, Elm...). You can even use tslint.json to define the guidelines which the team will agree upon, and engineers will be notified if they for example define a method without a return type.
+
+```json
+// tslint.json
+{
+    "rules": {
+        "no-any": true,
+        "triple-equals": true,
+        "prefer-const": true
+    }
+}
+```
+
+You can check all available tslint rules here - <https://palantir.github.io/tslint/rules/>
+
+Now that we have our simple tslint, let's see how it works:
+
+```ts
+// some-file.ts
+
+let me: any = "John Doe"; // Error - cannot use 'any', use 'string'
+
+me = false; // Error - cannot assign boolean to string variable.
+
+const age = 17;
+
+age = 18; // Error - cannot reassign const value
+
+let isEighteen = age == 18; // Warning - Use === instead of ==
+
+// Warning - `isEighteen` should be a const
+
+```
+
+You are probably thinking: "Ok well const is ES6, my browser will tell me i cannot reassign it". That is correct, but now, you will be notified at compile-time, meaning your app won't build until you fix the issue. Thus, we have strictly typed JavaScript app.
+
+It has all the OO principles that Java has (and many much more sophisticated), such as interfaces, abstract classes, enums, and many many more.
+
+If this catches your attention, I advise to visit https://www.typescriptlang.org/docs/handbook/basic-types.html for more detailed information.
+
+## Build Tools and Node
```

Comment:
This section can be shorten, too. As you will still probably have 200-300 word over limit. Regarding that, maybe you can remove all the examples from the text and to leave just descriptions of concepts.